src/main/webapp/resources/js/pages/projects/list/ProjectsTable.jsx

```diff
@@ -64,11 +65,16 @@ const reducer = (state, action) => {
  */
 export function ProjectsTable() {
   const [state, dispatch] = useReducer(reducer, initialState);
-  const [total, setTotal] = useState(undefined);
-  const [loading, setLoading] = useState(false);
-  const [projects, setProjects] = useState(undefined);
+  const [total, setTotal] = React.useState(undefined);
+  const [loading, setLoading] = React.useState(false);
+  const [projects, setProjects] = React.useState(undefined);
+  const [paginationOptions, setPaginationOptions] = React.useState(null);
 
-  useEffect(() => {
+  React.useMemo(() => {
```

Comment:
This should be:

```jsx
const paginationOptions = React.useMemo(() => {
  getPaginationOptions(total);
}, [total]);
```

Reply:
Updated in c96bd77d27718c9064e6535c7aac61dd11b3f80b